runelite-client/src/main/java/net/runelite/client/game/AgilityShortcut.java

```diff
@@ -0,0 +1,223 @@
+	ARCEUUS_ESSENSE_MINE_WEST(73, "Rock Climb", new WorldPoint(1742, 3853, 0), ImmutableSet.of(ROCKS_27984, ROCKS_27985), new Rectangle(1742, 3853, 10, 2)),
+	LAVA_DRAGON_ISLE_JUMP(74, "Stepping Stone", new WorldPoint(3200, 3807, 0), ImmutableSet.of(STEPPING_STONE_14918)),
+	REVENANT_CAVES_DEMONS_JUMP(75, "Jump", new WorldPoint(3199, 10135, 0), ImmutableSet.of(PILLAR_31561), new Rectangle(3199, 10135, 2, 2)),
+	REVENANT_CAVES_ANKOU_EAST(75, "Jump", new WorldPoint(3201, 10195, 0), ImmutableSet.of(PILLAR_31561), new Rectangle(3201, 10195, 2, 2)),
+	REVENANT_CAVES_ANKOU_NORTH(75, "Jump", new WorldPoint(3180, 10209, 0), ImmutableSet.of(PILLAR_31561), new Rectangle(3179, 10208, 2, 2)),
+	ZUL_ANDRA_ISLAND_CROSSING(76, "Stepping Stone", new WorldPoint(2156, 3073, 0), ImmutableSet.of(STEPPING_STONE_10663)),
+	SHILO_VILLAGE_STEPPING_STONES( 77, "Stepping Stones", new WorldPoint(2863, 2974, 0), ImmutableSet.of(STEPPING_STONE_16466)),
+	KHARAZI_JUNGLE_VINE_CLIMB(79, "Vine", new WorldPoint(2897, 2939, 0), ImmutableSet.of(NULL_26884, NULL_26886)),
+	TAVERLEY_DUNGEON_SPIKED_BLADES(80, "Strange Floor", new WorldPoint(2877, 9813, 0), ImmutableSet.of(STRANGE_FLOOR)),
+	SLAYER_DUNGEON_CHASM_JUMP(81, "Spiked Blades", new WorldPoint(2770, 10003, 0), ImmutableSet.of(STRANGE_FLOOR_16544)),
+	LAVA_MAZE_NORTH_JUMP(82, "Stepping Stone", new WorldPoint(3092, 3880, 0), ImmutableSet.of(STEPPING_STONE_14917)),
+	BRIMHAVEN_DUNGEON_EAST_STEPPING_STONES_NORTH(83, "Stepping Stones", new WorldPoint(2685, 9547, 0), ImmutableSet.of(STEPPING_STONE_19040)),
+	BRIMHAVEN_DUNGEON_EAST_STEPPING_STONES_SOUTH(83, "Stepping Stones", new WorldPoint(2693, 9529, 0), ImmutableSet.of(STEPPING_STONE_19040)),
+	ELVEN_ADVANCED_CLIFF_SCRAMBLE(85, "Rocks", new WorldPoint(2337, 3253, 0), ImmutableSet.of(ROCKS_16514, ROCKS_16514), new Rectangle(2332, 3251, 6, 3)),
+	KALPHITE_WALL(86, "Crevice", new WorldPoint(3214, 9508, 0), ImmutableSet.of(CREVICE_16465)),
+	BRIMHAVEN_DUNGEON_VINE_EAST(87, "Vine", new WorldPoint(2672, 9582, 0), ImmutableSet.of(VINE_26880, VINE_26882)),
+	BRIMHAVEN_DUNGEON_VINE_WEST(87, "Vine", new WorldPoint(2606, 9584, 0), ImmutableSet.of(VINE_26880, VINE_26882)),
+	REVENANT_CAVES_CHAMBER_JUMP(89, "Jump", new WorldPoint(3240, 10144, 0), ImmutableSet.of(PILLAR_31561), new Rectangle(3240, 10144, 2, 2));
+
+	@Getter
+	private final int level;
+	@Getter
+	private final String description;
+	@Getter
+	private final WorldPoint location;
+	@Getter
+	private final Set<Integer> obstacleIds;
+	@Getter
+	private final Rectangle shortcutBounds;
+
+	// Bounding rectangle is used in cases where multiple object IDs are on the same screen with different levels e.g. Cosmic altar
+	// Null locations won't show on the world map
+	AgilityShortcut(int level, String description, WorldPoint location, Set<Integer> obstacleIds, Rectangle bounds)
+	{
+		this.level = level;
+		this.description = description;
+		this.location = location;
+		this.obstacleIds = obstacleIds;
+		this.shortcutBounds = bounds;
```

Comment:
I still don't understand why bounds are required when we already have a worldlocation for the shortcut?

Reply:
I added the bounds for shortcuts with more than 1 obstacle ID, where there are duplicates in the same area. The Arceuus rune altar obstacles are all the same 2 rock obstacles, but in different locations and distance between them. The WorldLocation doesn't match with the obstacles directly - it lines up with the world map icon.

Reply:
The world map icons are not aligned with the actual obstacles?

Reply:
What if you use a Multimap and then calculate the distance between the obstacle and each map point, and take the one that is closest?